stack.go

```diff
@@ -111,6 +113,30 @@ func (c Call) Format(s fmt.State, verb rune) {
 		buf := [6]byte{}
 		s.Write(strconv.AppendInt(buf[:0], int64(line), 10))
 
+	case 'k':
+		pkg := c.fn.Name()
+
+		var fullPath string
+		var start, end int
+
+		const pathSep = "/"
+		const pkgSep = "."
+		if i := strings.LastIndex(pkg, pathSep); i != -1 {
+			start = (i + len(pathSep))
+			if end = strings.Index(pkg[i+len(pathSep):], pkgSep); end != -1 {
+				end += (i + len(pkgSep))
+				fullPath = pkg[:end]
+			}
+		}
+
+		if s.Flag('+') {
+			start = 0
+		}
+
+		pkg = fullPath[start:end]
```

Comment:
@ChrisHines agreed, that was an implementation error. This looks much cleaner and all of the tests pass. Re-pushing with this change.